lib/std/heap.zig

```diff
@@ -8,6 +8,141 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const page_size_os: ?usize = if (builtin.os.tag.isDarwin())
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
+        else => null,
+    }
+else if (builtin.os.tag == .windows)
+    // -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 4 << 10,
+        else => null,
+    }
+else
+    null;
+
+/// This value defines the largest page size for this architecture/OS combination that the standard library allows. The standard library asserts that the `pageSize()` does not exceed `page_size_max`. To allow larger page sizes, override `page_size_max` as well as `-z max-page-size`.
+pub const page_size_max: usize = page_size_os orelse switch (builtin.cpu.arch) {
```

Comment:
```suggestion
pub const max_page_size: usize = page_size_os orelse switch (builtin.cpu.arch) {
```

We generally prefer to put the `min`/`max` first (except in cases where we're mirroring C names).